8236912: NullPointerException when clicking in WebView with Button 4 or Button 5

effad opened the pull request:
As documented in JDK-8236912, WebView did not check whether the idMap really contained a mapping for the given button, making it prone to errors, when things are extended (as has happened here).

The fix consists of two test cases that show the problem in unfixed WebViews and a fix which works analogously to the check whether the given event type is mapped.
<!-- Anything below this marker will be automatically updated, please do not edit manually! -->
## Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

## Issue
[JDK-8236912](https://bugs.openjdk.java.net/browse/JDK-8236912): NullPointerException when clicking in WebView with Button 4 or Button 5


## Approvers
 * Guru Hb ([ghb](@guruhb) - **Reviewer**)
 * Kevin Rushforth ([kcr](@kevinrushforth) - **Reviewer**)

kevinrushforth commented:
The PR title should match the title of the bug report in JBS exactly. Can you change it to:

```
8236912: NullPointerException when clicking in WebView with Button 4 or Button 5
```

This will need two reviewers. I'll review it and also request @guruhb to review.

kevinrushforth commented:
@effad - The extra commits and merge are no problem, provided that you want to get this into openjfx15.

However, if you intend to get this into openjfx14, as currently targeted in JBS, then you will need to:

1. Rebase your `JDK-8236912` branch on top of the upstream `jfx14` branch
2. Force-push your branch: `git push --force origin JDK-8236912`
3. Edit this PR and change the target to the `jfx14` branch

kevinrushforth commented:
NOTE that this PR is not yet ready for integration. In addition to needing one more reviewer (me), there is the open question of which branch it should be targeted to.

@effad I think that this should go into openjfx14, so the above steps I listed will allow for this.

@guruhb can you please cancel your review (click the "re-request review" icon next to your review) until the above is done?

effad commented:
I've rebased my branch like this:
```
git fetch upstream
git rebase upstream/jfx14
git pull
git push
```
and changed the base branch of this PR to jfx14. Hope that was the correct way to do this ;-).

kevinrushforth commented:
Your branch still has commits from `master` after jfx14 was branched. This was the problem:
```
git pull
```

This pulled in additional changes from master that were already in your branch and effectively undid the rebase. You need to redo it, but this time instead of pulling changes after you rebase, you should simply do:
```
git push --force origin JDK-8236912
```

effad commented:
> ```
> git push --force origin JDK-8236912
> ```
I did 
```
git rebase upstream/jfx14
git push --force origin JDK-8236912
```
but that has messed up things further, didn't it?

kevinrushforth commented:
Oh, I see the problem. In addition to rebasing you need to exclude any commits that are from the `master` branch and are not your. So what you really need to do is:

```
git rebase -i upstream/jfx14
<EDIT THE LIST OF COMMITS AND DROP ANY THAT ARE NOT YOURS>
git push --force origin JDK-8236912
```

Sorry for not anticipating this problem.

effad commented:
> Sorry for not anticipating this problem.
I am sorry for not knowing git well enough. Thank you for your patience and help ;-). I think the PR now contains what it should ;-).

kevinrushforth commented:
Yes, it looks fine now, so it is ready for review. And this is one thing that can be a bit tricky about using git with multiple branches.

Btw, I was going to suggest `git rebase -i` the first time (it's what I always do in such a situation), but thought I could save you a step. Lesson learned. :)

guruhb commented:
+1, Looks good to me.

kevinrushforth commented:
@effad you can integrate this when ready. I will sponsor it.

effad commented:
/integrate

kevinrushforth commented:
/sponsor
